SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKEditor.cpp

```diff
@@ -84,3 +130,55 @@ void FSpatialGDKEditor::GenerateSnapshot(UWorld* World, FString SnapshotFilename
 		FailureCallback.ExecuteIfBound();
 	}
 }
+
+void FSpatialGDKEditor::RemoveEditorAssetLoadedCallback()
+{
+	if (OnAssetLoadedHandle.IsValid())
+	{
+		return;
+	}
+
+	if (UEditorEngine* EdEngine = Cast<UEditorEngine>(GEngine))
+	{
+		UE_LOG(LogSpatialGDKEditor, Log, TEXT("Removing UEditorEngine::OnAssetLoaded."));
+		FCoreUObjectDelegates::OnAssetLoaded.RemoveAll(EdEngine);
+		UE_LOG(LogSpatialGDKEditor, Log, TEXT("Replacing UEditorEngine::OnAssetLoaded with spatial version that won't run during schema gen."));
+		OnAssetLoadedHandle = FCoreUObjectDelegates::OnAssetLoaded.AddLambda([this](UObject* Asset) {
+			OnAssetLoaded(Asset);
+		});
+	}
+
+}
+
+// This callback is copied from UEditorEngine so that we can turn it off during schema gen in editor.
+void FSpatialGDKEditor::OnAssetLoaded(UObject* Asset)
+{
+	// do not init worlds when running schema gen.
+	if (bSchemaGeneratorRunning)
+	{
+		return;
+	}
+
+	UWorld* World = Cast<UWorld>(Asset);
+	if (World)
+	{
+		// Init inactive worlds here instead of UWorld::PostLoad because it is illegal to call UpdateWorldComponents while IsRoutingPostLoad
+		check(World);
```

Comment:
Is this `check` redundant, given the `nullptr` check of the block entry-point?

Reply:
Done